core/backend/src/SQLiteDb.ts

```diff
@@ -76,76 +77,61 @@ export class SQLiteDb implements IDisposable {
     this.nativeDb.closeDb();
   }
 
-  /** Returns true if the SQLiteDb is open */
+  /** Returns true if this SQLiteDb is open */
   public get isOpen(): boolean { return this.nativeDb.isOpen(); }
 
+  /** Returns true if this SQLiteDb is open readonly */
+  public get isReadonly(): boolean { return this.nativeDb.isReadonly(); }
+
   /**
-   * 1. open a database
-   * 2. call a function supplying the open SQliteDb as argument. If it is async, await its return.
-   * 3. close the database (even if exceptions are thrown)
+   * Open a database, perform an operation, then close the database.
+   *
+   * Details:
+   * - if database is open, throw an error
+   * - open a database
+   * - call a function with the database opened. If it is async, await its return.
+   * - if function throws, abandon all changes, close database, and rethrow
+   * - save all changes
+   * - close the database
+   * @return value from operation
    */
-  public static withOpenDb<T>(args: {
-    /** The name of the database to open */
-    dbName: string;
-    /** either an object with the open parameters or just OpenMode value. */
-    openMode?: OpenMode | SQLiteDb.OpenParams;
-    /** @internal */
-    container?: SQLiteDb.CloudContainer;
-  }, operation: (db: SQLiteDb) => T): T {
-    const db = new SQLiteDb();
-    db.openDb(args.dbName, args.openMode ?? OpenMode.Readonly, args.container);
-    let fromPromise = false;
+  public withOpenDb<T>(args: SQLiteDb.WithOpenDbArgs, operation: () => T): T {
```

Comment:
Have you changed the function signature of a `@public` function?

Reply:
yes, but it is new in the yet-to-be-released version. I'm not too worried about breaking anything, but it is poor form.

Reply:
It is about to be released in 3.3. This will break it in 3.4. If you want to change it, we should change it in 3.3 prior to release.

Reply:
not too late? I'll create a PR now then.